engine/script/src/script_sys.cpp

```diff
@@ -1201,12 +1201,12 @@ union SaveLoadBuffer
     * project root.
     *
     * @name sys.reboot
-    * @param [arg1] [type:string] argument 1
-    * @param [arg2] [type:string] argument 2
-    * @param [arg3] [type:string] argument 3
-    * @param [arg4] [type:string] argument 4
-    * @param [arg5] [type:string] argument 5
-    * @param [arg6] [type:string] argument 6
+    * @param [arg1] [type:string|number] argument 1
+    * @param [arg2] [type:string|number] argument 2
+    * @param [arg3] [type:string|number] argument 3
+    * @param [arg4] [type:string|number] argument 4
+    * @param [arg5] [type:string|number] argument 5
+    * @param [arg6] [type:string|number] argument 6
```

Comment:
The interface is a string.
I don't think we should tell users to pass in a number, when we use checkstring.
Also, this change is unrelated to the rest of the PR.
This is what we meant with keeping changes separate.
If we need to revert the PR, _everything_ will be reverted.

Please undo this from this PR.

Reply:
I "corrected" the documentation since I touched the function. But I get your point.

I added `number` since `lua_tostring` will convert number to string. Maybe the Defold team has discussed this and how to handle it in code and docs — I wanted to bring it up so I know what's deemed correct.

Personally, I prefer to have API documentation that is true to the behavior of the function, even if it makes the docs less concise. Silently converting number arguments may cause user confusion and issues for tool authors that parse the docs.

If the function should only accept strings, then it should call `luaL_checktype(L, i, LUA_TSTRING)` before `lua_tostring`.

Allowing implicit number-to-string and string-to-number argument conversions is arguably desirable, as it's expected in Lua. But it also creates situations where the user by mistake might pass a number to an engine function, which Lua will convert to a string, and the engine will then hash — that's not an insignificant amount of work to happen implicitly.

If you think this is a worthwhile discussion to have, I can open an issue about it.